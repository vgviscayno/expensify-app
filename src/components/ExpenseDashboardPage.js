import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpenseSummary from './ExpenseSummary';

const ExpenseDashboardPage = (props) => (
    <div>
        <ExpenseSummary />
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);

export default ExpenseDashboardPage;
