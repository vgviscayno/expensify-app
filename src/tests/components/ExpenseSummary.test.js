import { ExpenseSummary } from '../../components/ExpenseSummary';
import { shallow } from 'enzyme';
import React from 'react';
import expenses from '../fixtures/expenses';
import selectExpensesTotal from '../../selectors/expense-total';

test('should correctly render ExpenseSummary with 1 expense', () => {
    const wrapper = shallow(
        <ExpenseSummary
            expenseCount={[expenses[0]].length}
            expensesTotal={selectExpensesTotal([expenses[0]])}
        />
    );
    expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpenseSummary with multiple expenses', () => {
    const wrapper = shallow(
        <ExpenseSummary
            expenseCount={expenses.length}
            expensesTotal={selectExpensesTotal(expenses)}
        />
    );
    expect(wrapper).toMatchSnapshot();
});
