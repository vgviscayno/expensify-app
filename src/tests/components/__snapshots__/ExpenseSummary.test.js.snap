// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpenseSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling
     
    $45.66
  </h1>
</div>
`;

exports[`should correctly render ExpenseSummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling
     
    $1,185.66
  </h1>
</div>
`;
